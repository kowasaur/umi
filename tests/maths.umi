il void print(int32) "call void [mscorlib]System.Console::WriteLine(int32)"
il int32 +(int32, int32) "add"
il int32 -(int32, int32) "sub"
il int32 *(int32, int32) "mul"
il int32 /(int32, int32) "div"
il int32 %(int32, int32) "rem"

void main() {
    int32 plus = 9 + 10
    int32 minus = plus - 3
    int32 mul = 3 * 5
    int32 div = 75 / mul
    int32 mod = 25 % 3
    print(plus)
    print(minus)
    print(mul)
    print(div)
    print(mod)

    int32 nesting = (5 + (6 / 2)) * 3
    print(nesting)
}
