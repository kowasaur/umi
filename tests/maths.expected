19
16
15
5
1
24
